### DISCUSSION TITLE
Refactoring code using gpt-4

### DISCUSSION THREAD

**Discussion by unicomp21** [microsoft/semantic-kernel#1734] [Q&A]
@stephentoub if I want to refactor code, using gpt-4, which is larger than 32k tokens, how would I go about it w/ SK?  Is there a sample I should be looking at?  Or is this a bad idea to begin with?

I refactor code all the time these days via cut/paste into gpt-4, but am starting to hit the wall wrt amount of code I can reckon w/ due to 8k, 16k, and 32k limits in gpt-x.

**Comment by dnc-nl** [comment]
Hello all, basically the same question:

So, I'm creating an internal tool that allows us to check for bugs, and do internal technical documentation of our c#/angular projects. I think it's really important that the conversation understands the full context of what it is working with to see the big picture. With Chat GPT you could provide all the different cs files and thus create one big conversation with the full context.

How do you think you can work around this 'limitation' I'm looking at semantic kernel, but I'm not sure this will do the trick, if you start summarizing code it doesn't really make sense, or am I missing something here? Does anyone have an idea how to solve this core issue?

**Comment by unicomp21** [comment]
I'm starting to think it's all about creating a loop for the assistant to rapid fire compile attempts and code editing via function calls?

For larger codebases, checking for semantic bugs, I struggle w/ this one.  It's almost like we need a new modeling layer which plays nicely w/ gpt-4?

**Reply by dnc-nl** [reply]
It looks like chatgpt webpage is using the parameter "parent_message_id", but I can't seem to use this one within the API.

I think the most important thing is that the full context of the project (sln) and all code files attached are in the same context.

**Comment by markwallace-microsoft** [comment]
I am cleaning up old unanswered discussions
